build.gradle

```diff
@@ -0,0 +1,101 @@
+def jvmEncoding = java.nio.charset.Charset.defaultCharset().name()
+if (jvmEncoding != "UTF-8") {
+	throw new IllegalStateException("Build environment must be UTF-8 (it is: $jvmEncoding) - add '-Dfile.encoding=UTF-8' to the GRADLE_OPTS environment variable ")
+}
+
+if (!JavaVersion.current().java8Compatible) {
+	throw new IllegalStateException("Must be built with Java 8 or higher")
+}
+
+buildscript {
+	repositories {
+		jcenter()
+		mavenCentral()
+		maven {
+			url 'https://plugins.gradle.org/m2/'
+		}
+		mavenLocal()
+	}
+
+	dependencies {
+		classpath "gradle.plugin.com.google.protobuf:protobuf-gradle-plugin:0.8.0"
+		classpath "gradle.plugin.io.ratpack:ratpack-gradle:1.4.2"
+		classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.6'
+	}
+}
+
+repositories {
+	mavenLocal()
+	jcenter()
+	maven {
+		credentials {
+			username smartThingsUserName
+			password smartThingsPassword
+		}
+		url "https://smartthings.artifactoryonline.com/smartthings/libs-release-local"
+	}
+	maven {
+		credentials {
+			username smartThingsUserName
+			password smartThingsPassword
+		}
+		url "https://smartthings.artifactoryonline.com/smartthings/libs-snapshot-local"
+	}
+}
+
+group 'smartthings'
+version = rootProject.file('version.txt').text.trim()
+
+apply plugin: "base"
+apply plugin: 'java'
```

Comment:
Since you're including the groovy plugin don't think the java one is necessary.  I believe groovy is an extension of the java plugin.